gio-macros/src/actions.rs

```diff
@@ -0,0 +1,556 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use darling::{util::Flag, FromMeta};
+use proc_macro::TokenStream;
+use proc_macro2::{Span, TokenStream as TokenStream2};
+use quote::{format_ident, quote, quote_spanned};
+use std::collections::BTreeMap;
+use syn::{
+    parse, spanned::Spanned, Attribute, Error, FnArg, ImplItem, ImplItemMethod, ItemImpl, Lit,
+    Meta, MetaList, NestedMeta, PatType, Signature, Type,
+};
+
+#[derive(Debug, Default, FromMeta)]
+#[darling(default)]
+pub struct ActionImplAttributes {
+    register_fn: Option<String>,
+}
+
+#[derive(Debug, Default, FromMeta)]
+#[darling(default)]
+struct ActionAttributes {
+    name: Option<String>,
+    stateful: Flag,
+    initial_state: Option<Lit>,
+    change_state: Flag,
+    no_parameter: Flag,
+}
+
+#[derive(Debug)]
+struct ActionHandlerInfo {
+    attrs: ActionAttributes,
+    sig: Signature,
+}
+
+impl ActionHandlerInfo {
+    fn action_name(&self) -> String {
+        if let Some(ref name) = self.attrs.name {
+            name.clone()
+        } else {
+            self.sig.ident.to_string()
```

Comment:
```suggestion
            self.sig.ident.to_string().replace("_", "-")
```

The convention in GTK is to use dashes in action names, so I think this makes sense as a default? If that sounds good, a note can be added in the docs about this and you can mention it can still be changed manually to underscores with `#[action(name = "...")]`